include/seqan3/io/detail/misc.hpp

```diff
@@ -96,4 +97,65 @@ void set_format(format_variant_type & format,
         throw unhandled_extension_error("No valid format found for this extension.");
 }
 
+/*!\brief A helper unary type trait to test whether a file format has the required member `file_extensions`.
+ * \ingroup io
+ * \tparam format_t The format type to check.
+ * \returns Inherits from std::false_type if `format_t` is invalid, otherwise std::true_type.
+ */
+template <typename format_t>
+struct format_has_extension : public std::false_type
+{};
+
+//!\cond
+template <typename format_t>
+    requires (requires { format_t::file_extensions; })
+struct format_has_extension<format_t> : public std::true_type
+{};
+//!\endcond
+
+/*!\brief Checks if all formats in a type list have a static `file_extensions` member.
+ * \ingroup io
+ * \tparam    type_list_t The template template class holding the format types.
+ * \tparam    args_t...   A template parameter pack over the expected formats.
+ * \param[in] type_list   The type list containing the expected formats.
+ * \returns `true` if all formats are valid, otherwise `false`.
+ */
+template <template <typename ...> typename type_list_t, typename ... args_t>
+constexpr bool all_formats_have_file_extensions(type_list_t<args_t...> const & SEQAN3_DOXYGEN_ONLY(type_list))
+{
+    return (format_has_extension<args_t>::value && ... && true);
+}
+
+/*!\brief Returns a list of valid file extensions.
+ * \ingroup io
+ * \tparam formats_t The list of formats to parse; seqan3::detail::all_formats_have_file_extensions must return `true`.
+ * \returns `std::vector<std::string>` with all valid file extensions specified by `valid_formats`.
+ *
+ * ### Complexity
+ *
+ * Linear in the number of file extensions.
+ *
+ * ### Thread-safety
+ *
+ * thread-safe. Only reads from shared memory.
+ *
+ * ### Exception
+ *
+ * Strong exception guarantee. No input is modified. Might throw std::bad_alloc.
+ */
+template <typename formats_t>
+//!\cond
+    requires (all_formats_have_file_extensions(formats_t{}))
```

Comment:
Do you need this? Aren't the formats already required by concept to have the extensions static member?

Reply:
Do we have a concept for the formats? I thought not but I'll have a second look or you point me there?

Reply:
Ah ok, now I understand what you mean. Well, I thought since it is now a free function which could be potentially used otherwise it is good to check if the stuff added is what we want. But since the PR is quite old I did not checked this via a static assert. But I would keep it for this reasons but make it a static assert instead